### DISCUSSION TITLE
Testing out Table Many Rows Control - adding delete button?

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#176] [General]
Hi.

I have been having a lot of fun testing out the table controls.  It is really really interesting to be able to create an ON-CLICK event for a Table Many Row control and get the application to JUST refresh the table.  That is all kinds of cool, I am going to enjoy building asynchronous website behaviour.

Question:  I am testing out how the Table Many Rows control works and I cannot figure out how to add a button to each row to which I would like t attach a function, let's say delete.  I cannot figure out how to add a child item to the control and modify the output in any way.  Is this where we have to build custom queries our selves?

Note: The Table Many Rows control is only using this function:  (get-row target panel)

Screenshot of what I am working with: 
![contacts](https://user-images.githubusercontent.com/16431651/179425209-572c0614-d269-420a-a9f5-f3a88deda26f.png)

**Comment by rabbibotton** [comment]
Send my your .clog file

**Comment by aykaramba** [comment]
Okay, so:

1) I rebuilt the file using w3.css styling.

2) The events are:

On header:  
`(create-child target "<tr class='w3-green'><th>Row ID</th><th>First Name</th><th>Last Name</th><th>Country</th><th>DELETE ITEM</th></tr>")`

On click: (get-row target panel)

I have tried various permutations of on click to concatenate at the end of each row text or an image or a button, but nothing worked.  Ideally, it would be really interesting to see how the Table Many Rows control loop can be customized to modify the results on the fly.  I just want to know if or how it can be done.

3) SPECIAL NOTE: The Table Many Rows control has an on click event that refreshes the table and updates the results so I can see the form input working.

4) SPECIAL NOTE: For some reason, the Table May Rows on-header event does not render on load with the latest CLOG but it does once you click on the table.  I rebuilt the form from scratch using w3.css templating to get a 3 column layout, imported the html file and only added 3 db controls to the layout.  There is no strange nesting or css involved that would cause the rendering issue as far as I can see.

Here is the template01.clog file as per request:  

[template01.clog.tar.gz](https://github.com/rabbibotton/clog/files/9144202/template01.clog.tar.gz)

**Reply by rabbibotton** [reply]
will spend time on this soon

**Reply by aykaramba** [reply]
Greatly appreciate everything that you have built so far.  My humble requests may be wasting your time, feel free to set that aside if more important work awaits.  Thank you for the follow up.

**Reply by rabbibotton** [reply]
It is never a waste of my time, I need to exercise the code, bug detect, gain others perspective on what I have written, learn insights from their efforts even failures. You and others are gold.

**Comment by rabbibotton** [comment]
I am having a tough time following which means I didn't make this think easy enough to use or understand.

Let's start from scratch and I can then think how to make this all much easier.

Goal
contact records and a report - table of contacts with ability to delete per row?
